src/main/webapp/js/simulation_visualization.js

```diff
@@ -13,10 +15,14 @@ async function visualize() {
     var mainHeader = document.getElementById('simulation-visualization-header');
     var roundHeader = document.getElementById('current-round-header');
     var delay;
+    var pauseButton = document.getElementById('pause-button');
     mainHeader.innerText = 'Visualizing Simulation\n' + simulation.id;
     var params = {};
     params['simulationId'] = simulation.id;
     for (var round = 0; round < simulation.roundsNum; round++) {
+        while (pauseButton.innerHTML === "Play") {
```

Comment:
Consider adding a function ("isPaused") to wrap this condition - otherwise I find it hard to guess what the condition means.